app/services/fact_importer.rb

```diff
@@ -139,17 +137,16 @@ def add_new_facts
   def update_facts
     time = Time.now.utc
     updated = 0
-    db_facts_names = []
-    host.fact_values.includes([:fact_name]).joins(:fact_name).where(FACT_NAME_TYPE => fact_name_class_name).reorder('').find_each do |record|
-      next unless fact_names.include?(record.name)
-      new_value = facts[record.name]
```

Comment:
I wonder about this. Perhaps we're micro optimizing, but we're trading in a variable for an additional hash lookup if they differ. We also lose some readability. I'm not sure removing `new_value` is really worth it. Have you done some measurements on this?

Reply:
That's a valid question. I made a micro-benchmark and your intuition seems right. For the benchmark, I took the ansible facts fixture, which I ran through the initial importer to get it in the same structure. I ended up with a 400 facts hash and picked a random one to check for both true and false cases:

```ruby
def with_var_true(facts)
  a = facts['ansible_bios_version']
  if a == "1.00"
    return a
  end  
end  

def without_var_true(facts)
  if facts['ansible_bios_version'] == "1.00"  
    return facts['ansible_bios_version']
  end  
end  

def with_var_false(facts)
  a = facts['ansible_bios_version']  
  if a == "2.00"
    return a
  end  
end  

def without_var_false(facts)
  if facts['ansible_bios_version'] == "2.00"
    return facts['ansible_bios_version']
  end  
end  
```

```
[99] pry(main)> Benchmark.ips do |x|
[99] pry(main)*   x.report('without_var'){without_var_true(facts)}  
[99] pry(main)*   x.report('with_var'){with_var_true(facts)}  
[99] pry(main)*   x.report('with_var_false'){with_var_false(facts)}    
[99] pry(main)*   x.report('without_var_false'){without_var_false(facts)}    
[99] pry(main)*   x.compare!
[99] pry(main)* end  
Warming up --------------------------------------
         without_var   455.993k i/100ms
            with_var   659.015k i/100ms
      with_var_false   644.892k i/100ms
   without_var_false   639.280k i/100ms
Calculating -------------------------------------
    without_var_true      4.652M (± 2.6%) i/s -     23.256M in   5.002323s
       with_var_true      6.578M (± 1.8%) i/s -     32.951M in   5.011105s
      with_var_false      6.490M (± 0.9%) i/s -     32.889M in   5.068029s
   without_var_false      6.596M (± 1.1%) i/s -     33.243M in   5.040774s

Comparison:
     without_var_false:  6595537.2 i/s
         with_var_true:  6577737.3 i/s - same-ish: difference falls within error
        with_var_false:  6490106.8 i/s - same-ish: difference falls within error
      without_var_true:  4652251.0 i/s - 1.42x  (± 0.00) slower
```
In other words, while the false branch is pretty much the same in performance, the true branch is much slower without the extra variable. I'll revert this change.